client/service-container-chain/src/data_preservers.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright (C) Moondance Labs Ltd.
+// This file is part of Tanssi.
+
+// Tanssi is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Tanssi is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Tanssi.  If not, see <http://www.gnu.org/licenses/>
+
+use {
+    crate::spawner::{ContainerChainSpawner, TSelectSyncMode},
+    dc_orchestrator_chain_interface::{
+        DataPreserverAssignment, OrchestratorChainInterface, OrchestratorChainResult,
+    },
+    futures::stream::StreamExt,
+    std::{future::Future},
+    tc_consensus::ParaId,
+};
+
+pub type ProfileId = <dancebox_runtime::Runtime as pallet_data_preservers::Config>::ProfileId;
+
+async fn try_fut<T, E>(fut: impl Future<Output = Result<T, E>>) -> Result<T, E> {
+    fut.await
+}
+
+/// Watch assignements by indefinitly listening to finalized block notifications and switching to
+/// the chain the profile is assigned to.
+pub async fn task_watch_assignment<S: TSelectSyncMode>(
```

Comment:
it would be good if we can unitest this function. Do you think it is very difficult?

Reply:
I was going to recommend to use `handle_update_assignment` because that's already tested, but not sure if it can be easily integrated as that function is quite complex.

Reply:
@nanocryk how difficult would it be? what is your opinion?

Reply:
Not sure it is a good idea to merge it with `handle_update_assignment`, especially since `task_watch_assignment` is relatively short. I can take a look at how you test the other function to write similar tests :)

Reply:
fixed